app/controllers/JourneySetupController.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.AppConfig
+import javax.inject.{Inject, Singleton}
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc._
+import services.EnrolmentsAuthService
+import play.api.libs.json.JsValue
+import common.SessionKeys
+
+import scala.concurrent.Future
+
+@Singleton
+class JourneySetupController @Inject()(val messagesApi: MessagesApi,
+                                       val enrolmentsAuthService: EnrolmentsAuthService,
+                                       implicit val appConfig: AppConfig)
+  extends AuthorisedController with I18nSupport {
+
+  def journeySetup():Action[AnyContent] = authorisedAction {
+    implicit request =>
+    implicit agent =>
+    request.body.asJson match {
+
+      case Some(json) if extractRedirectUrl(json).nonEmpty => {
+        Future.successful(Ok("Received and stored the redirect url: " + extractRedirectUrl(json))).map {
```

Comment:
This needs testing as it's reducing the overall coverage below the minimum threshold